test/shlibloadtest.c

```diff
@@ -117,6 +117,7 @@ static int test_lib(void)
         if (!TEST_true(shlib_load(path_crypto, &cryptolib))
                 || !TEST_true(shlib_load(path_ssl, &ssllib)))
             goto end;
+        break;
```

Comment:
Ay, looks like gcc did not catch it because it was in false-conditional context.